code-challenges.test.js

```diff
@@ -13,40 +13,134 @@
 const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
-// b) Create the function that makes the test pass.
+const gotString = (string) => {
+    const codeMap = {
+      a: '4',
+      e: '3',
+      E: '3',
+      i: '1',
+      o: '0',
+    };
+    // finished this code using gpt , I came up to the '/[aeEio]/g ' part, didn't knew what to do to actually replace for the values I wanted, It was only replacing the 'a'
+    // with 4 .
+    return string.replace(/[aEeio]/g, (match) => codeMap[match])
+}
+console.log(gotString(secretCodeWord1))
+console.log(gotString(secretCodeWord2))
+console.log(gotString(secretCodeWord3))
+
+// Method 2:
+// const secretCodeWord1 = "Lackadaisical"
+// const secretCodeWord2 = "Gobbledygook"
+// const secretCodeWord3 = "Eccentric"
+// const gotString =  (string) => {
+//     let newStr = string.replace(/[aeio]/g, (match) => {
+//         switch (match) {
+//             case 'a':
+//               return '4'
+//             case 'e':
+//               return '3'
+//             case 'E':
+//               return '3'
+//             case 'i':
+//               return '1'
+//             case 'o':
+//               return '0'
+//         }
+//     })
+//     return newStr
+// }
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
-
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+describe('filterArray', () => {
+    it("should filter the array based on the given letter", () => {
+        const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+        const filterLetterA = "a"
+        const filterLetterE = "e"
+        
+        expect(filterArray(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+        // Expected output: ["Mango", "Apricot", "Peach"]
+        
+        expect(filterArray(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+        // Expected output: ["Cherry", "Blueberry", "Peach"]
+    })
+});
 
 // b) Create the function that makes the test pass.
+// pseudocode -
+// input -> array containing strings
+// output -> array containing specific letter in their strings
+// process -
+
+// create a function named 'filterArray'
+const filterArray = (arr, letter) => {
+// create a variable to store the new array with the expected letters
+let newArray = [];
+
+    // use a for loop to iterate through the array
+    for (let i=0; i < arr.length; i++){
+        // locate the letter we want using .includes
+        if (arr[i].includes(letter)){
```

Comment:
Great thought process, but what about the capital A in 'Apricot'? This currently won't push that word into the newArray.